react/features/talk-while-muted/middleware.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import { setCurrentNotificationUid } from './actions';
+import {
+    TALK_WHILE_MUTED_SOUND_FILE,
+    TALK_WHILE_MUTED_SOUND_ID
+} from './constants';
+
+import { APP_WILL_MOUNT, APP_WILL_UNMOUNT } from '../base/app';
+import { CONFERENCE_JOINED } from '../base/conference';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
+import { setAudioMuted } from '../base/media';
+import { MiddlewareRegistry } from '../base/redux';
+import { playSound, registerSound, unregisterSound } from '../base/sounds';
+import {
+    hideNotification,
+    showNotification
+} from '../notifications';
+
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+    const { dispatch, getState } = store;
+    const { conference } = action;
+
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        dispatch(registerSound(TALK_WHILE_MUTED_SOUND_ID, TALK_WHILE_MUTED_SOUND_FILE));
+        break;
+    case APP_WILL_UNMOUNT:
+        dispatch(unregisterSound(TALK_WHILE_MUTED_SOUND_ID));
+        break;
+
+    case CONFERENCE_JOINED: {
+        conference.on(
+            JitsiConferenceEvents.TRACK_MUTE_CHANGED,
+            track => {
+                const { currentNotificationUid } = getState()['features/talk-while-muted'];
+
+                if (currentNotificationUid && track.isAudioTrack() && track.isLocal() && !track.isMuted()) {
+                    dispatch(hideNotification(currentNotificationUid));
+                    dispatch(setCurrentNotificationUid());
+                }
+            });
+        conference.on(
+            JitsiConferenceEvents.TALK_WHILE_MUTED, () => {
+                const notification = dispatch(showNotification({
```

Comment:
Isn't it going to be safer to get the result directly from `showNotification` and than dispatch it?

I'm worried about async behaviour in the middlewares. Not sure what would be the result from the `dispatch` if we add async behaviour in a middleware in future.

WDYT?

Reply:
Ok, I will change it. But, I just moved the code.